applications/FluidDynamicsApplication/python_scripts/coupled_fluid_thermal_solver.py

```diff
@@ -0,0 +1,177 @@
+                "solver_type" : "ThermallyCoupled",
+                "fluid_solver_settings": {
+                        "solver_type": "navier_stokes_solver_vmsmonolithic",
+                        "model_import_settings": {
+                                "input_type": "mdpa",
+                                "input_filename": "unknown_name"
+                        }
+                },
+                "thermal_solver_settings": {
+                        "solver_type": "Transient",
+                        "analysis_type": "linear",
+                        "model_import_settings": {
+                                "input_type": "use_input_model_part",
+                                "input_filename": "unknown_name"
+                        },
+                        "computing_model_part_name": "Thermal",
+                        "material_import_settings": {
+                                "materials_filename": "ThermicMaterials.json"
+                        },
+                        "convection_diffusion_variables": {
+                                "density_variable": "DENSITY",
+                                "diffusion_variable": "CONDUCTIVITY",
+                                "unknown_variable": "TEMPERATURE",
+                                "volume_source_variable": "HEAT_FLUX",
+                                "surface_source_variable": "FACE_HEAT_FLUX",
+                                "projection_variable": "PROJECTED_SCALAR1",
+                                "convection_variable": "CONVECTION_VELOCITY",
+                                "mesh_velocity_variable": "MESH_VELOCITY",
+                                "transfer_coefficient_variable": "",
+                                "velocity_variable": "VELOCITY",
+                                "specific_heat_variable": "SPECIFIC_HEAT",
+                                "reaction_variable": "REACTION_FLUX"
+                        }
+                }
+            }
+            """)
+
+        ## Overwrite the default settings with user-provided parameters
+        self.settings = custom_settings
+        self.settings.ValidateAndAssignDefaults(default_settings)
+
+        import python_solvers_wrapper_fluid
+        self.fluid_solver = python_solvers_wrapper_fluid.CreateSolverByParameters(self.main_model_part, self.settings["fluid_solver_settings"],"OpenMP")
+        
+        
+        self.thermal_model_part =  KratosMultiphysics.ModelPart("thermal_model_part")
+        modeler = KratosMultiphysics.ConnectivityPreserveModeler()
+        modeler.GenerateModelPart(self.main_model_part, self.thermal_model_part, "Element2D3N", "Condition2D2N")
+        
+        import python_solvers_wrapper_convection_diffusion
+        self.thermal_solver = python_solvers_wrapper_convection_diffusion.CreateSolverByParameters(self.thermal_model_part,self.settings["thermal_solver_settings"],"OpenMP")
+        
+
+    def AddVariables(self):
+        self.fluid_solver.AddVariables()
```

Comment:
In the past, @jcotela and me agreed that each solver in the FluidDynamicsApplication should add its own variables (@jcotela correct me if I'm wrong). We are aware of the fact that if a new variable is added to one of the "father" solvers, it needs to be added in the coupled one as well. However, this avoids this type of hack solutions and makes easier to follow which variables are used in each solver.

Reply:
unfortunately this doesn t work here. the point is that omly the fluid imports the modelpart, so that the fluid-model_part variables must also contain the thermal ones.

Reply:
Now I see you point. In that way the coupled solver works with any fluid solver...